### DISCUSSION TITLE
bug about http(s) when give {{Hostname}}

### DISCUSSION THREAD

**Discussion by qianbenhyu** [projectdiscovery/nuclei#4584] [Q&A] [answered]
this is my poc
![image](https://github.com/projectdiscovery/nuclei/assets/32768810/5e3bc426-f615-42f4-9855-bfa05c49d282)
my command is
![image](https://github.com/projectdiscovery/nuclei/assets/32768810/02be100e-8bae-4281-9c0b-e10a6e7baaa1)
but the request is
![image](https://github.com/projectdiscovery/nuclei/assets/32768810/d1729bed-292d-413b-999d-1c186ee849e9)

i give a http request, but nuclei gives a https request.

**Comment by tarunKoyalwar** [comment, marked as answer]
@qianbenhyu i am not able reproduce this issue , this could most likely be due to server or requests are being upgraded by proxy

```console
$ nuclei -u http://localhost:8000 -t tmp.yaml               

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v3.1.3

		projectdiscovery.io

[INF] Current nuclei version: v3.1.3 (latest)
[INF] Current nuclei-templates version: v9.7.2 (latest)
[WRN] Scan results upload to cloud is disabled.
[INF] New templates added in latest release: 61
[INF] Templates loaded for current scan: 1
[WRN] Executing 1 unsigned templates. Use with caution.
[INF] Targets loaded for current scan: 1
[flow-redirect-extract] [http] [medium] http://localhost:8000/admin.html
```

and connection type `i.e http or https` is determined by input url (i.e -u flag input)
